### PR TITLE
Refactor twitter preview component and update share preview

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Related to #75426

## Proposed Changes

* Break `TwitterPreview` and `Tweet` class components into smaller TS functional components
* Update calypso share preview for Twitter to use the updated preview component
* Update linkedin preview to render the image only if present

### Prerequisites
- Make sure you have a Twitter account.
- Make sure you have a self-hosted site connected to Jetpack, with a subscription that includes Social.
- From the _Social_ or _My Jetpack_ page of your site WPadmin, activate Social.
- Then, connect your site to Twitter (check [these instructions](https://jetpack.com/support/jetpack-social/connecting-to-social-networks/) if you need help). You may need to create a test page.
- Spin up Calypso, by running this branch locally or using a live link below.

### Testing

- Create a new post from your site `wp-admin`
- Add a featured image (do not add a custom media)
- Publish it
- Visit `/posts/:site` in Calypso
- You should see the newly created post under the _Published_ tab
- In the post menu, click _Share_, then press the _Preview_ button
- Select the Twitter preview
- You should see a preview of how the post looks when shared

| BEFORE | AFTER |
|--------|--------|
| Screenshot 2023-04-20 at 3 35 33 PM |Screenshot 2023-04-20 at 3 34 40 PM |

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/social-previews/src/linkedin-preview/index.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ export function LinkedInPreview( {
 					/>
 				) : null }
 				<article>
-					<img className="linkedin-preview__image" src={ image } alt="" />
+					{ image ? <img className="linkedin-preview__image" src={ image } alt="" /> : null }
```
Comment (monsieur-z) [inline packages/social-previews/src/linkedin-preview/index.tsx:78]: Just for brevity, this could be `image && `

**Review by monsieur-z — APPROVED** [review]
Code looks good to me, and the PR tested as expected.

File: packages/social-previews/src/linkedin-preview/index.tsx (hunk 1 above)
Comment (manzoorwanijk) [reply packages/social-previews/src/linkedin-preview/index.tsx:78]: I try avoid `&&` as much as possible. The read is that we do not have runtime checks here like proptypes. So, if image is a falsy value like `0`, then `0` will be rendered which is not intended.

**Comment by noahtallen** [comment]
To fix the build issue that caused the revert, I think you just need to add this to `social-previews/package.json` under `devDependencies`: `"@types/wordpress__components": "^14.0.10",` (and then run yarn install from calypso's root)

**Pull request merged**